qemu/tests/qemu_guest_agent.py

```diff
@@ -181,14 +183,31 @@ def gagent_verify(self, params, vm):
         self.gagent.verify_responsive()
         logging.info(self.gagent.cmd("guest-info"))
 
+    @error_context.context_aware
     def setup(self, test, params, env):
         BaseVirtTest.setup(self, test, params, env)
         start_vm = params["start_vm"]
         if (not self.vm) and (start_vm == "yes"):
             vm = self.env.get_vm(params["main_vm"])
             vm.verify_alive()
             self.vm = vm
-            self.setup_gagent_in_guest(params, self.vm)
+
+            error_context.context("Check qemu-guest-agent status before install it", logging.info)
+            session = self._get_session(params, vm)
+            s, o = session.cmd_status_output(params.get("gagent_status_cmd"))
+            if s:
```

Comment:
There we have two conditions:
1. the qemu-ga service has not been installed at all.
2. the service was not running.

The service should just be installed for condition 1.

Reply:
yes, it is better to use two judgment conditions,one is used to check if installed, the other to check if started.
I modify it, pls have a check.